Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs

```diff
@@ -262,7 +262,13 @@ private Task ProcessMessageAsync(Node node, IncomingMessage message)
         /// <param name="message">The message payload.</param>
         private async Task SendMempoolPayloadAsync(Node node, MempoolPayload message)
         {
-            Guard.Assert(node == this.AttachedNode); // just in case
+            if (node != this.AttachedNode)
+            {
+                this.logger.LogError("Node Mismatch ({0}:'{1}', {2}:'{3}')", nameof(node), node?.RemoteSocketEndpoint, nameof(this.AttachedNode), this.AttachedNode?.RemoteSocketEndpoint);
```

Comment:
We should probably have a `Guard.NotNull(node, nameof(node));` in front of all these methods. We can then remove the conditonal access (`node?`).